src/app/+item-page/simple/item-types/shared/item.component.ts

```diff
@@ -0,0 +1,190 @@
+          relTypesCurrentPage[idx].rightLabel === label)
+        )
+      ),
+      distinctUntilChanged(compareArraysUsingIds())
+    );
+
+/**
+ * Operator for turning a list of relationships into a list of the relevant items
+ * @param {string} thisId           The item's id of which the relations belong to
+ * @param {ItemDataService} ids     The ItemDataService to fetch items from the REST API
+ * @returns {(source: Observable<Relationship[]>) => Observable<Item[]>}
+ */
+export const relationsToItems = (thisId: string, ids: ItemDataService) =>
+  (source: Observable<Relationship[]>): Observable<Item[]> =>
+    source.pipe(
+      flatMap((rels: Relationship[]) =>
+        observableZip(
+          ...rels.map((rel: Relationship) => {
+            let queryId = rel.leftId;
+            if (rel.leftId === thisId) {
+              queryId = rel.rightId;
+            }
+            return ids.findById(queryId);
+          })
+        )
+      ),
+      map((arr: Array<RemoteData<Item>>) =>
+        arr
+          .filter((d: RemoteData<Item>) => d.hasSucceeded)
+          .map((d: RemoteData<Item>) => d.payload)),
+      distinctUntilChanged(compareArraysUsingIds()),
+    );
+
+/**
+ * Operator for turning a list of relationships into a list of metadatarepresentations given the original metadata
+ * @param thisId      The id of the parent item
+ * @param itemType    The type of relation this list resembles (for creating representations)
+ * @param metadata    The list of original Metadatum objects
+ * @param ids         The ItemDataService to use for fetching Items from the Rest API
+ */
+export const relationsToRepresentations = (thisId: string, itemType: string, metadata: MetadataValue[], ids: ItemDataService) =>
+  (source: Observable<Relationship[]>): Observable<MetadataRepresentation[]> =>
+    source.pipe(
+      flatMap((rels: Relationship[]) =>
+        observableZip(
+          ...metadata
+            .map((metadatum: any) => Object.assign(new MetadataValue(), metadatum))
+            .map((metadatum: MetadataValue) => {
+            if (metadatum.isVirtual) {
+              const matchingRels = rels.filter((rel: Relationship) => ('' + rel.id) === metadatum.virtualValue);
+              if (matchingRels.length > 0) {
+                const matchingRel = matchingRels[0];
+                let queryId = matchingRel.leftId;
+                if (matchingRel.leftId === thisId) {
+                  queryId = matchingRel.rightId;
+                }
+                return ids.findById(queryId).pipe(
+                  getSucceededRemoteData(),
+                  map((d: RemoteData<Item>) => Object.assign(new ItemMetadataRepresentation(itemType), d.payload))
+                );
+              }
+            } else {
+              return of(Object.assign(new MetadatumRepresentation(itemType), metadatum));
+            }
+          })
+        )
+      )
+    );
+
```

Comment:
Done

Reply:
thanks for creating utils file. there is a reason why `relationsToRepresentations` was not moved to `item-relationships-utils.ts` ?

Reply:
Likely @Atmire-Kristof didn't move it because it wasn't used anywhere else.

I've moved it now in https://github.com/DSpace/dspace-angular/pull/390/commits/17636906db93ea8fa565068ab25da6b87568eb0d